app/src/main/java/chat/rocket/android/authentication/twofactor/presentation/TwoFAPresenter.kt

```diff
@@ -44,9 +44,13 @@ class TwoFAPresenter @Inject constructor(private val view: TwoFAView,
                         try {
                             // The token is saved via the client TokenProvider
                             val token = client.login(usernameOrEmail, password, twoFactorAuthenticationCode)
-                            multiServerRepository.save(server, TokenModel(token.userId, token.authToken))
-                            registerPushToken()
-                            navigator.toChatList()
+                            if (token != null) {
+                                multiServerRepository.save(server, TokenModel(token.userId, token.authToken))
```

Comment:
I'm not sure but I think if token happens to be null it will fall through the catch to `RocketChatException`. :thinking:

Reply:
Whit the changes made on the [RestResult](https://github.com/filipedelimabrito/Rocket.Chat.Kotlin.SDK/blob/d71e51b5b566a8d6762c29201fdc3eee52702dcf/core/src/main/kotlin/chat/rocket/core/internal/rest/RestResult.kt) and [RestMultiResult.kt](https://github.com/filipedelimabrito/Rocket.Chat.Kotlin.SDK/blob/d71e51b5b566a8d6762c29201fdc3eee52702dcf/core/src/main/kotlin/chat/rocket/core/internal/rest/RestMultiResult.kt) class we expected null results because `fromJson` can return null.

Reply:
We should never expect null results from the SDK, I was going to comment on those changes... If the SDK doesn't return a value, it should throw...

Reply:
Are you saying that if the result from two mentioned above class is null (for `result`, `update` and `remove`) we should throw an exception? Is it? I'll make changes there so. 

As soon as you do the review, I'll look into to merge those PRs as quickly as possible to deliver the features..

Reply:
I commented on the SDK PR, the calls should be nonnullable.